src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -85,6 +97,48 @@ public void addSection(Section inputSection) {
         }
     }
 
+    public void deleteSection(Long stationId) {
+        checkOnlyOneSectionInLine();
+        checkContainsStationInLine(stationId);
+
+        Section firstSection = findFirstSection();
+        if (firstSection.equalsUpStation(stationId)) {
+            sections.remove(firstSection);
+            return;
+        }
+
+        Section lastSection = findLastSection();
+        if (lastSection.equalsDownStation(stationId)) {
+            sections.remove(lastSection);
+            return;
+        }
+
+        Section frontSection = sections.stream().filter(it -> it.equalsDownStation(stationId)).findFirst().get();
```

Comment:
스트림은 행위단위로 내려쓰는 것이 가독성에 더 좋겠네요.
그리고 내부에도 `it` 보다는 의미있는 네이밍을 주면 좋을 것 같아요.

```suggestion
        Section frontSection = sections.stream()
                .filter(it -> it.equalsDownStation(stationId))
                .findFirst()
                .orElseThrow(() -> new RuntimeException("구간을 찾지 못했습니다."));
```

Reply:
넵! 감사합니다! section 으로 했다가 외부에서 section 을 사용하는 곳이 있어서 고민을 하다가 it 로 해버렸던 것 같아요 ㅠㅠ   다른 이름 생각해볼게요!